src/FixedDataTableBufferedRows.js

```diff
@@ -77,119 +77,96 @@ class FixedDataTableBufferedRows extends React.Component {
       this._staticRowArray.length = rowsToRender.length;
     }
 
-    var baseOffsetTop = props.offsetTop - props.scrollTop;
-
-    for (let i = 0; i < rowsToRender.length; i++) {
-      const rowIndex = rowsToRender[i];
-
-      // if the row doesn't exist in the buffer, assign a fake row to it.
-      // this is so that we can get rid of unnecessary row mounts/unmounts
+    // render each row from the buffer into the static row array
+    for (let i = 0; i < this._staticRowArray.length; i++) {
+      let rowIndex = rowsToRender[i];
+      // if the row doesn't exist in the buffer set, then take the previous one
       if (rowIndex === undefined) {
-        // if a previous row existed, let's just make use of that
-        if (this._staticRowArray[i] === undefined) {
-          this._staticRowArray[i] = this.getStubRow(i, true, -1);
-        }
-        continue;
+        rowIndex = this._staticRowArray[i] && this._staticRowArray[i].props.index;
       }
 
-      const currentRowHeight = this.props.rowSettings.rowHeightGetter(rowIndex);
-      const currentSubRowHeight = this.props.rowSettings.subRowHeightGetter(rowIndex);
-      const rowOffsetTop = baseOffsetTop + props.rowOffsets[rowIndex];
-      const rowKey = props.rowKeyGetter ? props.rowKeyGetter(rowIndex) : i;
-      const hasBottomBorder = (rowIndex === props.rowSettings.rowsCount - 1) &&
-        props.showLastRowBorder;
-      const visible = this.isRowInsideViewport(rowIndex);
-
-      this._staticRowArray[i] =
-        <FixedDataTableRow
-          key={rowKey}
-          isScrolling={props.isScrolling}
-          index={rowIndex}
-          width={props.width}
-          height={currentRowHeight}
-          subRowHeight={currentSubRowHeight}
-          rowExpanded={props.rowExpanded}
-          scrollLeft={Math.round(props.scrollLeft)}
-          offsetTop={Math.round(rowOffsetTop)}
-          visible={visible}
-          fixedColumns={props.fixedColumns}
-          fixedRightColumns={props.fixedRightColumns}
-          scrollableColumns={props.scrollableColumns}
-          onClick={props.onRowClick}
-          onContextMenu={props.onRowContextMenu}
-          onDoubleClick={props.onRowDoubleClick}
-          onMouseDown={props.onRowMouseDown}
-          onMouseUp={props.onRowMouseUp}
-          onMouseEnter={props.onRowMouseEnter}
-          onMouseLeave={props.onRowMouseLeave}
-          onTouchStart={props.onRowTouchStart}
-          onTouchEnd={props.onRowTouchEnd}
-          onTouchMove={props.onRowTouchMove}
-          showScrollbarY={props.showScrollbarY}
-          className={joinClasses(
-            rowClassNameGetter(rowIndex),
-            cx('public/fixedDataTable/bodyRow'),
-            cx({
-              'fixedDataTableLayout/hasBottomBorder': hasBottomBorder,
-              'public/fixedDataTable/hasBottomBorder': hasBottomBorder,
-            })
-          )}
-        />;
+      this._staticRowArray[i] = this.renderRow({
+        rowIndex,
+        key: i,
```

Comment:
out of interest, why is this the key not set to `rowIndex`?

Reply:
because if the key is given as a function of rowIndex, rows will get mounted/unmounted whenever row indexes change. So we keep the keys constant, form 0 to no:rows present.

FDT uses IntegerBufferSet to manage row index to key mappings, so that when you scroll and it leads to a new row getting visible in the viewport, it'll replace the old one that went outside the viewport. 

So basically if we scroll down by 1 row, for an initial viewport of 10 rows, then row 0 at key 0 might gets replaced by row 10, and the key should still be 0 to prevent react mounting/unmounting.